test/unit/shim/message-shim.test.js

```diff
@@ -976,6 +983,27 @@ tap.test('MessageShim', function (t) {
       })
     })
 
+    t.test('should properly time promise based consumers', function (t) {
```

Comment:
@michaelgoin can you explain why this works just fine without any intervention? I see we're intercepting promise [here](https://github.com/newrelic/node-newrelic/blob/main/lib/shim/message-shim.js#L803) but we aren't binding the segment so I don't see how it's doing the proper touch of the active segment in this case.

Reply:
oh i think it's probably because it does `tx.end()` in the handler for interceptPromise which appears to touch the base segment

Reply:
Ah yeah, I think you are right.

Looks like in this case we are dealing with a base segment which the transaction end calls...

```
if (this.baseSegment) {
    this.baseSegment.touch()
  }
```

For a non-'baseSegment', either the timing wouldn't get updated (if already updated once) or it would get its timing updated and name changed to `Truncated/<the name>`.

It may just be a case of luck here haha.